Source/SwiftLintFramework/Rules/Idiomatic/StaticOperatorRule.swift

```diff
@@ -79,23 +78,34 @@ public struct StaticOperatorRule: ASTRule, ConfigurationProviderRule, OptInRule
         ]
     )
 
-    public func validate(file: SwiftLintFile, kind: SwiftDeclarationKind,
-                         dictionary: SourceKittenDictionary) -> [StyleViolation] {
-        guard kind == .functionFree,
-            let offset = dictionary.offset,
-            let name = dictionary.name?.split(separator: "(").first.flatMap(String.init) else {
-                return []
-        }
+    public func makeVisitor(file: SwiftLintFile) -> ViolationsSyntaxVisitor? {
+        Visitor(viewMode: .sourceAccurate)
+    }
+}
 
-        let characterSet = CharacterSet(charactersIn: name)
-        guard characterSet.isDisjoint(with: .alphanumerics) else {
-            return []
+private extension StaticOperatorRule {
+    final class Visitor: SyntaxVisitor, ViolationsSyntaxVisitor {
+        private(set) var violationPositions: [AbsolutePosition] = []
+
+        override func visitPost(_ node: FunctionDeclSyntax) {
+            if node.isFreeFunction, node.isOperator {
+                violationPositions.append(node.funcKeyword.positionAfterSkippingLeadingTrivia)
+            }
         }
+    }
+}
 
-        return [
-            StyleViolation(ruleDescription: Self.description,
-                           severity: configuration.severity,
-                           location: Location(file: file, byteOffset: offset))
-        ]
+private extension FunctionDeclSyntax {
+    var isFreeFunction: Bool {
+        parent?.is(CodeBlockItemSyntax.self) ?? false
```

Comment:
My usual recommendation. 😄

Instead of checking for the parent subsequently, prefer to skip function declarations contained in struct, class, ... declarations by stopping AST traversal right when they are encountered.

Reply:
my concern is that we might miss one of these cases - we can do it as an optimization, but I wouldn't want to rely on that